gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/auth/GcsDelegationTokens.java

```diff
@@ -58,6 +62,11 @@
   /** Active Delegation token. */
   private Token<DelegationTokenIdentifier> boundDT;
 
+  private DelegationTokenInstantiationStrategy tokenInstantiationStrategy =
+      DelegationTokenInstantiationStrategy.INSTANCE_PER_SERVICE;
+  private static ConcurrentHashMap<Optional<String>, Token<DelegationTokenIdentifier>>
```

Comment:
Is there a reason that we use nullable String as a renewer/key? Seems like in Hadoop code base they use `Text` object for renewer: https://github.com/apache/hadoop/blob/02f6bad1fff0f9c8e8dc5899272a93ab64d15eb9/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/delegation/S3ADelegationTokens.java#L418

Reply:
Existing code related to GCS delegation tokens already uses string as a renewer. 

Later it is converted to `Text` when DelegationTokenIdentifier is created - https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/auth/AbstractDelegationTokenBinding.java#L123.

I do not have context on that decision.

Reply:
I think that it's better to switch code to use `Text` as soon as possible - in this case we will not need `Optional` wrapper for key values.